tests/v2/validation/fleet/provisioning/networkisolation.go

```diff
@@ -0,0 +1,231 @@
+package provisioning
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"regexp"
+	"strings"
+	"time"
+
+	"github.com/rancher/fleet/pkg/apis/fleet.cattle.io/v1alpha1"
+	"github.com/rancher/rancher/tests/v2/actions/kubeapi/namespaces"
+	"github.com/rancher/rancher/tests/v2/actions/kubeapi/workloads/deployments"
+	"github.com/rancher/rancher/tests/v2/actions/projects"
+	"github.com/rancher/rancher/tests/v2/actions/workloads/pods"
+	"github.com/rancher/shepherd/clients/rancher"
+	steveV1 "github.com/rancher/shepherd/clients/rancher/v1"
+	"github.com/rancher/shepherd/extensions/charts"
+	"github.com/rancher/shepherd/extensions/defaults"
+	"github.com/rancher/shepherd/extensions/kubeconfig"
+	"github.com/rancher/shepherd/extensions/unstructured"
+	"github.com/rancher/shepherd/pkg/namegenerator"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	kwait "k8s.io/apimachinery/pkg/util/wait"
+
+	extensionpods "github.com/rancher/shepherd/extensions/workloads/pods"
+)
+
+const defaultNamespace = "default"
+
+// testPNI is a set of in-order instructions that gets the IP from a pod in an existing deployment/project/namespace,
+// then creates a new deployment in the default project/namespace which tries to ping the existing pod.
+// This errors if the new pod can ping the existing one, meaning that PNI is not working.
+// Not designed to work with a deployment already in the default namespace!
+func testPNI(client *rancher.Client, clusterName, deploymentNamespace, deploymentName string) error {
+	// get one of the fleet deployment's pod IPs to use in the ping command later
+	fleetPodNames, err := pods.GetPodNamesFromDeployment(client, clusterName, deploymentNamespace, deploymentName)
+	if err != nil {
+		return err
+	}
+
+	firstPodObject, err := pods.GetPodByName(client, clusterName, deploymentNamespace, fleetPodNames[0])
+	if err != nil {
+		return err
+	}
+
+	corePod := &corev1.Pod{}
+	err = steveV1.ConvertToK8sType(firstPodObject, corePod)
+	if err != nil {
+		return err
+	}
+
+	// create a new workload / pod to ping that's hardened-compliant
+	falseP := false
```

Comment:
can you rename them to something like the below instead of falseP and trueP, so its easy to understand. 
allowPrivilegeEscalation := false
runAsNonRoot := true